### PR TITLE
test(DTFS2-7675): add facility value journey e2e test

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
We need e2e test coverage for each of the amendments journeys. This starts the journey for the user changing only the faciluty value

## Resolution :heavy_check_mark:
List all changes made to the codebase.

## Miscellaneous :heavy_plus_sign:
List any additional fixes or improvements.

### CODE REVIEW

**Review by BethThomas141 — DISMISSED** [review]
LGTM👍

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    whatDoYouNeedToChange.selectionErrorMessage().should('be.visible');
+    whatDoYouNeedToChange.selectionErrorMessage().contains('Select if you need to change the facility cover end date, value or both');
+  });
+
+  it('should submit change facility value', () => {
+    cy.visit(relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/what-do-you-need-to-change`));
+
+    whatDoYouNeedToChange.facilityValueCheckbox().click();
+    cy.clickContinueButton();
+  });
```
Comment (rkontogianni) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:80]: I think here you have to add cy.url().should('eq', relative(`something/facility-value`));  ? It might be better change the title of this assertion to navigate correctly to facility-value page.

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    cy.clickContinueButton();
+
+    facilityValue.errorSummary().should('be.visible');
+    facilityValue.errorSummary().contains('Enter the new facility value in number format');
+
+    facilityValue.facilityValueInlineError().should('be.visible');
+    facilityValue.facilityValueInlineError().contains('Enter the new facility value in number format');
+  });
+});
```
Comment (rkontogianni) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:93]: Could you add an assertion if facility value is too high ?

Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:93]: We've got unit tests to cover this - I dont think e2e tests should go into that level of granularity

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -955,6 +955,7 @@ jobs:
             './tfm/cypress/e2e/journeys/users/**/*.spec.js',
             './tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/**/*.spec.js',
             './ukef/cypress/e2e/journeys/portal/deal-cancellation/**/*.spec.js',
+            './ukef/cypress/e2e/journeys/portal/amendments/**/*.spec.js',
```
Comment (abhi-markan) [inline .github/workflows/test.yml:958]: Saw your reason in the description, my understanding is `tfm-facilities` get populated with mock data loader?

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    });
+  });
+
+  after(() => {
+    cy.clearCookies();
+    cy.clearSessionCookies();
+  });
+
+  it('should navigate through the journey correctly', () => {
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/what-do-you-need-to-change`));
```
Comment (abhi-markan) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js:61]: I know we have component test, but we can please assert some DOM elements on the page before clicking. Thanks

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    whatDoYouNeedToChange.facilityValueCheckbox().click();
+    cy.clickContinueButton();
+
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/facility-value`));
+    cy.keyboardInput(facilityValue.facilityValue(), '10000');
+    cy.clickContinueButton();
+
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/eligibility`));
```
Comment (abhi-markan) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js:69]: Assert some DOM elements when visitng EC page.

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import relative from '../../../relativeURL';
```
Comment (abhi-markan) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:1]: Nice to see a separate unhappy path spec.

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    cy.clearCookies();
+    cy.clearSessionCookies();
+  });
+
+  beforeEach(() => {
+    cy.clearSessionCookies();
+    cy.login(BANK1_MAKER1);
+  });
+
+  it('should render an error if nothing is selected to change', () => {
```
Comment (abhi-markan) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:63]: Same as above please render some DOM elements.

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    cy.clearSessionCookies();
+  });
+
+  it('should navigate through the journey correctly', () => {
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/what-do-you-need-to-change`));
+    whatDoYouNeedToChange.facilityValueCheckbox().click();
+    cy.clickContinueButton();
+
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/facility-value`));
+    cy.keyboardInput(facilityValue.facilityValue(), '10000');
```
Comment (abhi-markan) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js:66]: Can I see some more example of the facility value, I know there is a good unit test coverage but we still should run this on the E2E in case there is/are any breaking change(s) to the component or design library.

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js (hunk 2 above)
Comment (abhi-markan) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:95]: Thanks @MarRobSoftwire, we do need more test assertions on unhappy paths with edge cases, empty, not number and etc.

I have described the reason above on why it is a good approach to also cover this in an E2E tests assertions.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (MarRobSoftwire) [reply .github/workflows/test.yml:958]: nothing gets put in tfm by mock data loader

When I tried to make this test in e2e-tests/gef the deal wasnt being submitted properly (The e2e tests seem to jsut set the deal status to `confirmed` and dont actually send it to tfm) The amendments are stored in tfm-facilities so this breaks & the service returns a 404

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js (hunk 5 above)
Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.happy-path.spec.js:69]: EC page doesnt exist yet - this journey is incomplete

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js (hunk 2 above)
Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:95]: I've added e2e tests for every validation error, I think these will take much longer to run & be harder to maintain but will leave them in if this is what you have in mind

File: e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js (hunk 7 above)
Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/amendments/change-facility-value-journey.unhappy-path.spec.js:63]: Is that necessary in the unhappy path tests - the other test is checking everything renders correctly already

**Review by abhi-markan — CHANGES_REQUESTED** [review]

**Review by abhi-markan — APPROVED** [review]

**Review by rkontogianni — APPROVED** [review]
